rust/src/http2/http2.rs

```diff
@@ -0,0 +1,1175 @@
+                    //we check for completeness first
+                    if rem.len() < hl {
+                        //but limit ourselves so as not to exhaust memory
+                        if hl < HTTP2_MAX_HANDLED_FRAME_SIZE {
+                            return AppLayerResult::incomplete(
+                                (il - input.len()) as u32,
+                                (HTTP2_FRAME_HEADER_LEN + hl) as u32,
+                            );
+                        } else if rem.len() < HTTP2_MIN_HANDLED_FRAME_SIZE {
+                            return AppLayerResult::incomplete(
+                                (il - input.len()) as u32,
+                                (HTTP2_FRAME_HEADER_LEN + HTTP2_MIN_HANDLED_FRAME_SIZE) as u32,
+                            );
+                        } else {
+                            self.set_event(HTTP2Event::LongFrameData);
+                            self.request_frame_size = head.length - (rem.len() as u32);
+                        }
+                    }
+
+                    //get a safe length for the buffer
+                    let (hlsafe, complete) = if rem.len() < hl {
+                        (rem.len(), false)
+                    } else {
+                        (hl, true)
+                    };
+
+                    if head.length == 0 && head.ftype == parser::HTTP2FrameType::SETTINGS as u8 {
+                        input = &rem[hlsafe..];
+                        continue;
+                    }
+                    let txdata = self.parse_frame_data(
+                        head.ftype,
+                        &rem[..hlsafe],
+                        complete,
+                        head.flags,
+                        STREAM_TOSERVER,
+                    );
+
+                    let tx = self.find_or_create_tx(&head, &txdata, STREAM_TOSERVER);
+                    tx.handle_frame(&head, &txdata, STREAM_TOSERVER);
+                    let over = head.flags & parser::HTTP2_FLAG_HEADER_END_STREAM != 0;
+                    let txid = tx.tx_id;
+                    let ftype = head.ftype;
+                    tx.frames_ts.push(HTTP2Frame {
+                        header: head,
+                        data: txdata,
+                    });
+                    if ftype == parser::HTTP2FrameType::DATA as u8 {
+                        self.stream_data(STREAM_TOSERVER, &rem[..hlsafe], over, txid);
+                    }
+                    input = &rem[hlsafe..];
+                }
+                Err(nom::Err::Incomplete(_)) => {
+                    //we may have consumed data from previous records
+                    if input.len() < HTTP2_FRAME_HEADER_LEN {
+                        return AppLayerResult::incomplete(
+                            (il - input.len()) as u32,
+                            HTTP2_FRAME_HEADER_LEN as u32,
+                        );
+                    } else {
+                        panic!("HTTP2 invalid length frame header");
```

Comment:
can this only be the result of a bug or do we need more graceful handling here? Probably best to do a debug-validate bug with a return `AppLayerResult::err()`.

Reply:
This is meant to be unreachable code.
But this should be caught by the TCP checks about `AppLayerResult::incomplete` 
So I remove it